rasa/core/training/story_conflict.py

```diff
@@ -242,6 +255,47 @@ def _find_conflicting_states(
     }
 
 
+def _unpredictable_action(event: Event) -> bool:
+    """Identifies if the action cannot be predicted by policies that use story data.
+
+    Args:
+        event: An event to be checked.
+
+    Returns:
+        A boolean indicating if the event can be predicted or not.
+    """
+    return event.__str__() == ACTION_UNLIKELY_INTENT_NAME
+
+
+def _find_unpredictable_actions(
+    trackers: List[TrackerWithCachedStates],
+    domain: Domain,
+    max_history: Optional[int],
+    tokenizer: Optional[Tokenizer],
+) -> Dict[int, Optional[List[Text]]]:
+    """Identifies all states that contain actions that cannot be predicted.
+
+    Args:
+        trackers: Trackers that contain the states.
+        domain: The domain object.
+        max_history: Number of turns to take into account for the state descriptions.
+        tokenizer: A tokenizer to tokenize the user messages.
+
+    Returns:
+        A dictionary mapping state-hashes to a list of actions that cannot be predicted.
+    """
+    # Create a 'state -> list of actions' dict, where the state is
+    # represented by its hash
+    state_action_mapping = defaultdict(list)
+    for element in _sliced_states_iterator(trackers, domain, max_history, tokenizer):
+        hashed_state = element.sliced_states_hash
+        current_hash = hash(element.event)
+        if _unpredictable_action(element.event):
+            state_action_mapping[hashed_state] += [current_hash]
```

Comment:
Why does it have to store a list of action-hashes instead of a list of the actual action events? 🤔

Reply:
Just to make it compatible with the existing function called `_build_conflicts_from_states` and to make it possible to create `StoryConflict` instances.

Reply:
@alwx Sorry my review comment from this morning didn't make it through for some reason. 

What about doing as Anca said and iterating over `tracker.events` directly but then return a `StoryConflict` object right away instead of the hashed states? We could add this story conflict to the list of conflicts in line 184. We could also subclass `StoryConflict` to override its string representation as the current warning for `action_unlikely_intent` doesn't seem super helpeful. This would give us simpler code + better warnings.